### PR TITLE
add support for updating webhooks in user account

### LINKED ISSUE #3115: jx upgrade ingress update environment webhooks looks at wrong GitHub org
### Summary 

When updating ingress configuration via the `jx upgrade ingress` command, it does not update the environment webhooks. I believe this goes wrong because it uses the wrong GitHub organization.

As user and organization I've set `joostvdg` and that all works. When I do `jx upgrade ingress` and select `Y` on `Do you want to update all existing webhooks?` I get a drop down of every *other* GitHub organization I have access to.

```bash
Updating Jenkins with new external URL details https://jenkins.jx.jx.kearos.net
Previous webhook endpoint https://jenkins.jx.jx.kearos.net/github-webhook/
Updated webhook endpoint https://jenkins.jx.jx.kearos.net/github-webhook/
? Do you want to update all existing webhooks? Yes

Updating all webHooks from https://jenkins.jx.jx.kearos.net/github-webhook/ to https://jenkins.jx.jx.kearos.net/github-webhook/
? Which organisation do you want to use?  [Use arrows to move, type to filter]
> CatAndMonster
  FlussoBV
  abnamrocoesd
  cloudbees
```

My currently configured organisation, `joostvdg`, does not show up in the list.

### Steps to reproduce the behavior

Create a JX cluster in GKE:

```bash
jx create cluster gke \                                                                                                                                              
    --cluster-name="${JX_CLUSTER_NAME}" \
    --default-admin-password="${JX_ADMIN_PSS}" \
    --domain="${JX_DOMAIN}" \
    --kubernetes-version="${JX_K8S_VERSION}" \
    --machine-type='n1-standard-2' \
    --max-num-nodes=3 \
    --min-num-nodes=2 \
    --project-id=${JX_GCE_PROJECT} \
    --zone="${JX_K8S_ZONE}" \
    --skip-login
```

Use a custom domain, in this case, `jx.kearos.net`.
Then update ingress configuration via `jx upgrade ingress`, install cert-manager, use production lets encrypt, update webhooks to tls, and recreate all the ingress rules.

### Expected behavior

I get a dropdown with my current GitHub organization pre-selected. I select it and the webhooks get updated.

### Actual behavior

Nothing happens. There is not even a mention of the repositories not being found.

```bash
Previous webhook endpoint http://jenkins.jx.jx.kearos.net/github-webhook/
Updated webhook endpoint https://jenkins.jx.jx.kearos.net/github-webhook/
? Do you want to update all existing webhooks? Yes

Updating all webHooks from http://jenkins.jx.jx.kearos.net/github-webhook/ to https://jenkins.jx.jx.kearos.net/github-webhook/
? Which organisation do you want to use? demomon
```

Here I selected an organisation, which does exist but doesn't hold the environment repositories.
And after this, line, there's nothing, no webhook update, no complain about the repository not existing, nothing.

### Jx version

The output of `jx version` is:

```
NAME               VERSION
jx                 1.3.885
jenkins x platform 0.0.3402
Kubernetes cluster v1.11.7-gke.4
kubectl            v1.13.3
helm client        v2.12.3+geecf22f
helm server        v2.12.2+g7d2b0c7
git                git version 2.17.2 (Apple Git-113)
Operating System   Mac OS X 10.14.2 build 18C54
```

### Jenkins type

- [X] Classic Jenkins
- [ ] Serverless Jenkins

### Kubernetes cluster

GKE, see above.

### Operating system / Environment

MacOS -> `Mac OS X 10.14.2 build 18C54`

### PR DESCRIPTION
**Pull request by joostvdg**
#### Submitter checklist

- [X] Change is code complete and matches issue description.
- [ ] Change is covered by existing or new tests.

#### Description

This allows both the direct `update webhooks` and `upgrade ingress` commands to update webhooks belonging to either an organisation or to a user account.

In `upgrade ingress` it makes sure to also check for the current user as an option in the selection for where to look for updating webhooks.

In the GItHub implementation for the repo retrieval, it does a check to see if the org is the same as the user.  If it is the same, we should retrieve the list of repositories via the user-based API instead.

#### Special notes for the reviewer(s)

In summary, this allows people to update webhooks for a GitHub account rather than org.
This is sort-of "patched" in with a trick. Maybe it would be better to do a more thorough change for adding this kind of support. Or, add a warning this is not supported for accounts and people should use organisations instead - I might be the only sucker use Jenkins X like this.

#### Which issue this PR fixes

fixes #3115

### CODE REVIEW

**Comment by pmuir** [comment]
Can you rebase?

**Comment by joostvdg** [comment]
@pmuir rebased, let me know if this was done correctly. 
Might have time this weekend to see if I can add a test, but cannot guarantee.

**Comment by joostvdg** [comment]
@rawlingsj tests seem to fail unrelated to the code change (this time)?
Also, what failed last time with the BDD and Integration tests? I cannot see the details.

**Comment by joostvdg** [comment]
Requesting a retest, not sure if I can trigger these?

/test all
